scripts/process-reboot-cause

```diff
@@ -69,6 +71,57 @@ def read_reboot_cause_files_and_save_state_db():
                 x = TIME_SORTED_FULL_REBOOT_FILE_LIST[i]
                 os.remove(x)
 
+def get_sorted_reboot_cause_files(dpu_history_path):
+    """Retrieve and sort the reboot cause files for a specific DPU."""
+    try:
+        # Assuming you have a way to list the files in the directory
+        files = os.listdir(dpu_history_path)
+        # Filter and sort the files based on your criteria (e.g., by modification time)
+        sorted_files = sorted(
+            [os.path.join(dpu_history_path, f) for f in files if f.endswith('.txt')],
+            key=os.path.getmtime,  # Sort by modification time
+            reverse=True  # Most recent first
+        )
+        return sorted_files
+    except Exception as e:
+        sonic_logger.log_error(f"Error retrieving reboot cause files for {dpu_history_path}: {e}")
+        return []
+
+
+def read_dpu_reboot_cause_files_and_save_chassis_state_db():
+    """Retrieve reboot cause from history files and save them to chassisStateDB."""
+    chassis_state_db = swsscommon.SonicV2Connector(host="redis_chassis.server", port=6380)
```

Comment:
@rameshraghupathy can we have port# as a #define global variable?

Reply:
@prgeor Done